lumos/optimal_control/utils.py

```diff
@@ -297,23 +297,34 @@ def get_global_var_index(self, name: str) -> int:
     def get_global_var(self, x, name: str) -> float:
         return x[self.get_global_var_index(name)]
 
-    def get_var_index_in_dec(self, group: str, name: str, stage: int) -> int:
+    def get_var_index_in_dec(
+        self, group: str, name: str, stage: Optional[int] = None
+    ) -> Union[int, List[int]]:
         """Return the index of a variable in the decision var vector."""
-        if stage > self.num_stages - 1 or stage < -self.num_stages:
-            raise ValueError(
-                (
-                    f"stage must be bewteen [{-self.num_stages}, {self.num_stages-1}], "
-                    f"but got {stage}"
-                )
-            )
 
-        # allow negative index like python list
-        stage %= self.num_stages
+        if stage is None:
+            # Stage not given, then provide it for all stages using recursion
 
-        return (
-            self._stage_var_enum[self._make_stage_var_name(group=group, name=name)]
-            + stage * self.num_var_stage
-        )
+            return [
```

Comment:
should we revert to a more efficient implementation instead of using recursion? and then use recursion-ish method in test as ground truth

Reply:
done